src/Features/VisualBasic/Portable/Structure/Providers/CommentTriviaStructureProvider.vb

```diff
@@ -0,0 +1,22 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.PooledObjects
+Imports Microsoft.CodeAnalysis.Structure
+Imports Microsoft.CodeAnalysis.Text
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Structure
+    Friend Class CommentTriviaStructureProvider
+        Inherits AbstractSyntaxTriviaStructureProvider
+
+        Public Overrides Sub CollectBlockSpans(document As Document,
+                                               trivia As SyntaxTrivia,
+                                               spans As ArrayBuilder(Of BlockSpan),
+                                               cancellationToken As CancellationToken)
+            If trivia.Kind = SyntaxKind.CommentTrivia Then
+                VisualBasicOutliningHelpers.CollectCommentsRegions(trivia.Token.LeadingTrivia, spans)
+                VisualBasicOutliningHelpers.CollectCommentsRegions(trivia.Token.TrailingTrivia, spans)
```

Comment:
when would VB have outlining for trailing trivia?

Reply:
@CyrusNajmabadi When it is comment.

Reply:
When it is a comment.

Reply:
@CyrusNajmabadi 
An example containing both.
```vbnet
Case 0
 ' Trailing
 Console.WriteLine("Hello")
 ' Leading
Case 1
```
Which is the reason the vb implementation is more complex.

Reply:
I'm confused by your example.  The 'trailing' portion is leading trivia of 'console'.  It should not be trailing trivia at all.

Reply:
```
Imports System
Public Class C
    Public Sub M()
        ' C1
        Console.WriteLine("E1") 'C2
        Console.WriteLine("E2") 'C3
        ' C4
        Console.WriteLine("E3") 'C4
        Console.WriteLine("E4") 'C5
        ' C6
    End Sub
End Class
```
`C1, C4. C6` are leading all the rest are trailing.

Reply:
None of this is answering my question though :)  When do you need outlining for trailing trivia?

Reply:
This outlining provide "code folds" contiguous comment sections.